xtuner/v1/train/trainer.py

```diff
@@ -881,6 +891,41 @@ def _set_deterministic(self):
     def _set_random_seed(self, seed: int):
         set_random_seed(seed)
 
+    def _try_bind_numa(self):
+        if not HAS_NUMA:
+            logger.info("numa module not found, skip numa binding.")
+            return
+
+        try:
+            numa_node_num = numa.info.get_max_node() + 1
+            # get total gpu number of current node
+            nvsmi = nvidia_smi.getInstance()
+            total_GPU_per_node = len(nvsmi.DeviceQuery("memory.total")["gpu"])
```

Comment:
Should we use torch device count here to get GPU per node? If `CUDA_VISIBLE_DEVICES` is set, will this get an error number?